### PR TITLE
Add qoi

### PR DESCRIPTION
**Pull request by Julianiolo**
### Summary
Changes to recipe:  **qoi/1.0.0**

#### Details
I'm unsure on what to use as the version. The library does not seem to have an official versioning. The specification however is 1.0 ([see](https://qoiformat.org/qoi-specification.pdf)).

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by toge — APPROVED** [review]

**Comment by AbrilRBS** [comment]
> The library does not seem to have an official versioning. The specification however is 1.0 ([see](https://qoiformat.org/qoi-specification.pdf)).

@Julianiolo do you think we could ask upstream to generate a tag? Maybe they just have not thought about the usefulness of doing so for package managers

**Comment by Julianiolo** [comment]
@AbrilRBS I don't know if the intend on having versions, but we could probably ask them?

**Comment by Julianiolo** [comment]
@AbrilRBS I mean I kinda don't think they are planning on having any changes, as the specification stands, and the code is pretty minimal. The Readme and tests are updating more often. So I think either just calling it 1.0, or going with the commit hash might be the best idea, as there probably won't be any other versions soon.

But you can surely also ask them, I just don't know how active the owner is.

**Comment by jcar87** [comment]
@ team - no need to wait for a tag.

The upstream readme mentions existing packages in well-known repositories
https://github.com/phoboslab/qoi/?tab=readme-ov-file#packages

We need a version scheme that monotonically increases
* https://aur.archlinux.org/pkgbase/qoi-git
* https://packages.debian.org/bookworm/source/qoi

`0.x.cci.date` as previously discussed should be as good as any, and is more or less the approach by Debian

**Comment by Julianiolo** [comment]
@jcar87 Thanks for the answer!

How do you mean `0.x.cci.date` exactly? I assume `x` and `date` will be replaced by something?

File: recipes/qoi/all/test_package/test_package.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#define QOI_IMPLEMENTATION
+#include "qoi.h"
+
+int main(void) {
```
Comment (jcar87) [inline recipes/qoi/all/test_package/test_package.c:8]: I would simplify the implementation by a lot - we don't intend to test functionality at all, just integration that the library can be included.

`#include "qoi.h"` -> if this works it tends to be enough

we may want to refer to a symbol or something defined, for example we can print or access the contents of `qoi_padding` - that's enough for the test.

Comment (Julianiolo) [reply recipes/qoi/all/test_package/test_package.c:8]: @jcar87 yeah, I might have done a little overboard on that one... 😅

**Review by jcar87 — APPROVED** [review]
Thanks @Julianiolo !!

**Review by toge — APPROVED** [review]

File: recipes/qoi/all/test_package/test_package.c (hunk 1 above)
Comment (jcar87) [reply recipes/qoi/all/test_package/test_package.c:6]: Thanks so much for addressing it! :)

**Pull request merged**